### PR TITLE
Splash screen

### PR DESCRIPTION
**Pull request by Robadob**
Add SplashScreen whilst simulation loads, this mostly covers RTC time.

If executed via pyflamegpu, pyflamegpu logo is used instead of the regular flamegpu logo.
The window icon is also overridden, in Windows testing from C this also updates the task bar icon, but not from Python. Unclear why.
Updated image IO to use DevIL instead of stb_image
Added a class OverlayGroup for grouping linked Overlays
Changes to support SDL_image

Second commit in this PR updates screenshot writing to use DevIL

Related main repo PR: https://github.com/FLAMEGPU/FLAMEGPU2/pull/485

### CODE REVIEW

**Comment by Robadob** [comment]
Tried to test most recent commit with python, but getting a runtime error.

```
C:\Users\Robadob\fgpu2_vis\src\texture\Texture.cpp(262): Texture::loadImage(): File 'C:\Users\Robadob\AppData\Local\Temp\fgpu2\vis\6948867642060026471_pyflamegpu.png' could not be loaded!
 SDL_Image error: Failed loading libpng16-16.dll: The specified module could not be found.
 ```
 
 SDL_lmage appears unable to runtime load it's dependent dlls.
 
 The function used is defined here: https://github.com/libsdl-org/SDL/blob/5753fd73c39a87e7f836bdebfea7a7776958dbec/src/loadso/windows/SDL_sysloadso.c#L33
 
 But I can't see an obvious reason it would fail, the dlls appear to be in path etc. Need to investigate why winapi `LoadLibrary` isn't finding it. Perhaps a full rebuild would fix this?

**Comment by Robadob** [comment]
Discussion of the issue here: https://bugs.python.org/issue32516

A few potential solutions none of which I like.

A python lib which wraps SDL/SDL_image uses this: https://github.com/dholth/pysdl2-cffi/blob/9b7f21bb442ca82e6e282a944d494b0a5615d594/_sdl/helpers.py#L7

Quick testing has shown adding the dll's directory to `os.environ[`PATH']` does not fix it.

Erring on the side of moving to a different image loading sln.

**Comment by Robadob** [comment]
Splash screen should now be working on Windows for both C and py.
DevIL support may need tweaking for Linux CMake.

I essentially updated the image load method to load with `DevIL` and convert to an `SDL_Surface`.

File: src/texture/Texture.cpp [hunk 1]
```diff
@@ -253,42 +296,162 @@ void Texture::applyOptions() {
...
     GL_CALL(glBindTexture(type, 0));
 }
-std::shared_ptr<Texture::ImageData> Texture::findLoadImage(const std::string &imagePath) {
+std::shared_ptr<SDL_Surface> Texture::findLoadImage(const std::string &imagePath) {
     // Attempt without appending extension
-    std::shared_ptr<ImageData> image = loadImage(imagePath, false, true);
-    for (unsigned int i = 0; i < sizeof(IMAGE_EXTS) / sizeof(char*) && !image; i++) {
+    std::shared_ptr<SDL_Surface> image = loadImage(imagePath, false, true);
+    const char * IMAGE_EXTS[] = {"png", "jpg", "jpeg", "bmp", "tiff"};
+    for (int i = 0; i < sizeof(IMAGE_EXTS) / sizeof(char*) && !image; i++) {
```
Comment (ptheywood) [inline src/texture/Texture.cpp:303]: warning under linux, `i` should be an unsigned type
```
warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
```

**Comment by Robadob** [comment]
ty will fix tonight.

**Pull request merged**